sklearn/model_selection/_search.py

```diff
@@ -563,6 +563,20 @@ def inverse_transform(self, Xt):
         self._check_is_fitted('inverse_transform')
         return self.best_estimator_.inverse_transform(Xt)
 
+    @property
+    def n_features_in_(self):
+        # For consistency with other estimators we raise a AttributeError so
+        # that hasattr() fails if the search estimator isn't fitted.
+        try:
+            check_is_fitted(self)
+        except NotFittedError as nfe:
+            raise AttributeError(
+                "{} object has no n_features_in_ attribute."
+                .format(self.__class__.__name__)
+            ) from nfe
+
+        return self.best_estimator_.n_features_in_
```

Comment:
we could set `n_features_in_` in `fit`, and not have the property.

Reply:
Sure. I don't mind too much. I think for the searches we use properties a lot.

Reply:
at least in its current state, `classes_` is the only other public property.

Even `best_score_` seems not to be a property. It's just that setting it in
fit is less code and kinda cleaner?

Reply:
I was referring to all the `if_delegate_has_method` methods but yeah these aren't properties.
OK I'll set it in fit

Reply:
Thinking about it again I'd rather not: the best_estimator_ might just not have the attribute and that would raise an error.